batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/controller/StudentController.java

```diff
@@ -0,0 +1,201 @@
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * @Author binodnme
+ * Created on 1/14/16
+ */
+@WebServlet("/students/*")
+public class StudentController extends HttpServlet{
+    StudentService studentService = new StudentService();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException{
+        String servletPath = req.getPathInfo();
+
+        if(servletPath == null || "/".equalsIgnoreCase(servletPath)){
+            Integer page = 1;
+            try{
+                page = Integer.parseInt(req.getParameter("page"));
+            }catch (NumberFormatException e){}
+
+            list(req, resp, page);
+
+        }else{
+            String[] params = servletPath.split("/");
+
+            if(params[1].equalsIgnoreCase("create")){
+                create(req, resp);
+
+            } else if(params[2].equalsIgnoreCase("edit")){
+                Integer studentId = TypeCaster.toInt(params[1]);
+                edit(req, resp,studentId);
+            }
+        }
+
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String servletPath = req.getPathInfo();
+        String[] params = servletPath.split("/");
+
+        if(params[1].equalsIgnoreCase("create")){
+            createProcess(req,resp);
+
+        }else if(params[2].equalsIgnoreCase("edit")){
+            Integer studentId = TypeCaster.toInt(params[1]);
+            editProcess(req, resp, studentId);
+        }else if(params[2].equalsIgnoreCase("delete")){
+            Integer studentId = TypeCaster.toInt(params[1]);
+            if((studentId == null)){
+
+            }else{
+                deleteProcess(req, resp, studentId);
+            }
+        }
+    }
+
+
+    private void list(HttpServletRequest req, HttpServletResponse resp, int page) throws ServletException, IOException {
+        try {
+            List<Student> studentList = studentService.fetch(page);
+            req.setAttribute("studentList", studentList);
+            req.setAttribute("nextPageNum", page+1);
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/students.jsp").forward(req, resp);
+        } catch (DataException e) {
+            e.printStackTrace();
+            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            req.setAttribute("errorMessage", e.getMessage());
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(req, resp);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/WEB-INF/views/error.jsp" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)